### PR TITLE
HBW-44 Fix dependency usage in mixin

### PR DESCRIPTION
**Pull request by sugarfree1**

### CODE REVIEW

File: hbw/app/assets/javascripts/hbw_dependencies/noconflict.js.coffee [hunk 1]
```diff
@@ -9,12 +9,12 @@
 #= require react
 #= require ./wrap_globals
 
-HBWDependencies = ['$', 'jQuery', 'moment', 'React']
+HBWDependencies = ['$', 'jQuery', 'moment', 'React', 'define']
```
Comment (flash-gordon) [inline hbw/app/assets/javascripts/hbw_dependencies/noconflict.js.coffee]: I'd like to know we do we need this

Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw_dependencies/noconflict.js.coffee]: jQuery plugins which we use in HBW have algorithm for fetching current jQuery. For example select2.js:
```
function (factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['jquery'], factory);
  } else if (typeof exports === 'object') {
    // Node/CommonJS
    factory(require('jquery'));
  } else {
    // Browser globals
    factory(jQuery);
  }
}
```
as you see, they firstly try to import from require.js.

It hurts a lot when integrated system has require.js under the hood (like CRM). In this case plugins for jQuery try to import jQuery from CRM's require.js.

Comment (flash-gordon) [inline hbw/app/assets/javascripts/hbw_dependencies/noconflict.js.coffee]: then we must **remove** `require.js` before loading the plugin

Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw_dependencies/noconflict.js.coffee]: Yes, that is what my changes are all about. http://stackoverflow.com/questions/24746175/how-can-i-disable-require-js-at-runtime

**Comment by sugarfree1** [comment]
Updated

**Pull request merged**